app/services/katello/event_monitor/poller_thread.rb

```diff
@@ -81,18 +81,18 @@ def poll_for_events
         @thread.kill if @thread
         @thread = Thread.new do
           begin
-            Rails.application.executor.wrap do
-              @logger.info("Polling Katello Event Queue")
-              ::Katello::EventQueue.reset_in_progress
-              loop do
+            @logger.info("Polling Katello Event Queue")
+            ::Katello::EventQueue.reset_in_progress
+            loop do
+              Rails.application.executor.wrap do
```

Comment:
This fixes a deadlock when reloading code within the request cycle such as when hammer requests will issue a reload of the application code. @ianballou thanks for pointing this one out. Rails' DebugLocks middleware helped me see that my executor.wrap was not in the right place! https://api.rubyonrails.org/classes/ActionDispatch/DebugLocks.html